R/ff_importance.R

```diff
@@ -27,38 +28,21 @@
 #' }
 #'
 #' @export
-ff_importance <- function(model_path, output_csv, append = TRUE) {
-  # Check if model file exists
-  if (!file.exists(model_path)) {
-    stop("Model file does not exist: ", model_path)
+ff_importance <- function(model, output_csv, name = NA, append = TRUE) {
+  if (!has_value(name)) {
+    if (is.character(model)) {
+      name <- sub("\\.model$", "", basename(model))
+    }
   }
-
-  # Construct path for .rda file
-  rda_path <- sub("\\.model$", ".rda", model_path)
-
-  # Check if .rda file exists
-  if (!file.exists(rda_path)) {
-    stop("Corresponding .rda file does not exist: ", rda_path)
-  }
-
-  # Load model
-  model <- xgboost::xgb.load(model_path)
-
+  model <- load_model(model)
```

Comment:
is it clear where does this load_model comes from? it is not part of this file!

Reply:
it is a subfunction in ff_predict. What do you think, should i do it like this, copy the same subfunction to ff_run (and rename it i guess because otherwise there are two functions with the same name) or should i make a separate file with all functions that are used in different functions?

Reply:
I would suggest create a separate file like util or something and put all the shared functions in it